docs/src/HOWTOs/MsgSetNew.tla

```diff
@@ -0,0 +1,75 @@
+------------------------------- MODULE MsgSetNew ------------------------------
+
+VARIABLE 
+    \* @type: [ int: Set([x: Int]), str: Set([y: Str]) ];
+    msgs, 
+    \* @type: Bool;
+    found3,
+    \* @type: Bool;
+    foundC
+
+
+Ints == {1,2,3}
+Strs == {"a","b","c"}
+
+\* no annotation required
+Messages == [ 
+              int |-> [x: Ints],
+              str |-> [y: Strs]
+            ]
+
+Init == 
+    /\ msgs = [ int |-> {}, str |-> {} ]
+    /\ found3 = FALSE
+    /\ foundC = FALSE
+
+\* @type: ([x: Int]) => Bool; 
+SendInt(m) == 
+    msgs' = [msgs EXCEPT !.int = msgs.int \cup {m}]
+
+\* @type: ([x: Int]) => Bool; 
+RmInt(m) == 
+    msgs' = [msgs EXCEPT !.int = msgs.int \ {m}]
+
+\* @type: ([y: Str]) => Bool;
+SendStr(m) == 
+    msgs' = [msgs EXCEPT !.str = msgs.str \cup {m}]
+
+\* @type: ([x: Int]) => Bool; 
+RmStr(m) == 
+    msgs' = [msgs EXCEPT !.str = msgs.str \ {m}]
+
+
+AddIntMsg == 
+    /\ \E v \in Ints: 
+        /\ SendInt( [x |-> v] )
+    /\ UNCHANGED <<found3, foundC>>
+
+CheckIntMsg == 
+    /\ \E m \in msgs.int:
+        /\ found3' = ( m.x = 3 )
+        /\ RmInt(m)
+    /\ UNCHANGED foundC
+
+AddStrMsg == 
+    /\ \E v \in Strs: 
+        /\ SendStr( [y |-> v] )
+    /\ UNCHANGED <<found3, foundC>>
+
+CheckStrMsg == 
+    /\ \E m \in msgs.str:
+        /\ foundC' = ( m.y = "c" )
+        /\ RmStr(m)
+    /\ UNCHANGED found3
+
+Next ==
+    \/ AddIntMsg
+    \/ CheckIntMsg
+    \/ AddStrMsg
+    \/ CheckStrMsg
+
+TypeOk == 
+    /\ msgs.int \subseteq Messages.int
+    /\ msgs.str \subseteq Messages.str
```

Comment:
`msgs` is not a set, so `msgs \subseteq X` won't work. Any other implementation with `msgs \in X` would have to invoke `SUBSET` somewhere in `X`

Reply:
Ah yes. I figured that out and didn't mean to post this comment actually. Sorry.

This is another place where it would be nice to have syntax sugar eventually.